opus/application/static_media/js/browse.js

```diff
@@ -424,6 +464,50 @@ var o_browse = {
         });
     }, // end browse behaviors
 
+    // update order arrows right away when user clicks on sorting arrows in pill or table header
+    // sync up arrows in both sorting pill and table header
+    updateOrderIndicator: function(headerOrderIndicator, pillOrderIndicator, isDescending, slug) {
+        let headerOrder = isDescending ? "sort-desc" : "sort-asc";
+        let headerOrderArrow = isDescending ? tableSortUpArrow : tableSortDownArrow;
+        let isPillOrderDesc = isDescending ? "true" : "false";
+        let pillOrderTooltip = isDescending ? "Change to ascending sort" : "Change to descending sort";
+        let pillOrderArrow = isDescending ? pillSortUpArrow : pillSortDownArrow;
+
+        // If header already exists, we update the header arrow, else we do nothing
+        if (headerOrderIndicator && headerOrderIndicator.length !== 0) {
+            headerOrderIndicator.data("sort", `${headerOrder}`);
+            headerOrderIndicator.attr("class", `column_ordering ${headerOrderArrow}`);
+
+            // Reset arrows on rest of table headers
+            // let headers = $(`.dataTable th a:not([data-slug="opusid"], [data-slug=${slug}])`).find("span:last");
+            let headers = $(`.dataTable th a:not([data-slug=${slug}])`).find("span:last");
+            headers.data("sort", "none");
+            headers.attr("class", defaultTableSortArrow);
+        }
+
+        // Re-render each pill
+        let listHtml = "";
+        $.each(opus.prefs.order, function(index, orderEntry) {
+            isPillOrderDesc = orderEntry[0] === "-" ? "true" : "false";
```

Comment:
Although it works in this case, I think it's generally a bad idea to overwrite function parameters, because then if someone tries to use them later in the function they will be surprised they aren't the values what were passed in. It would be better to use new variables with local "let" scope inside the $.each.

Reply:
I think you missed a couple of cases: You reuse isPillOrderDesc and pillOrderArrow, overwriting the parameters that were passed in. I think these need to be new variables too. Also I'm not sure why you renamed this variable to "order". I would think of "order" as being the entire order slug, not just this one entry. Perhaps call it "orderEntrySlug", since that's what it really is?